gen/ldap_it4i

```diff
@@ -0,0 +1,103 @@
+our $A_MEMBER_STATUS;  *A_MEMBER_STATUS =  \'urn:perun:member:attribute-def:core:status';
+
+our $STATUS_VALID;      *STATUS_VALID =        \'VALID';
+our $STATUS_EXPIRED;    *STATUS_EXPIRED =      \'EXPIRED';
+our $STATUS_INVALID;    *STATUS_INVALID =      \'INVALID';
+our $STATUS_DISABLED;   *STATUS_DISABLED =     \'DISABLED';
+
+# check facility attribute
+my $ldapBaseDN = $data->getFacilityAttributeValue( attrName => $A_F_BASE_DN );
+if (!defined($ldapBaseDN)) {
+	exit 1;
+}
+
+# gather user data
+my $users;
+
+foreach my $resourceId ( $data->getResourceIds() ) {
+	foreach my $memberId ( $data->getMemberIdsForResource( resource => $resourceId )) {
+
+		my $login = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_LOGIN_EINFRA );
+		$users->{$login}->{$A_FIRST_NAME} = $data->getUserAttributeValue( member => $memberId, attrName => $A_FIRST_NAME );
+		$users->{$login}->{$A_LAST_NAME} = $data->getUserAttributeValue( member => $memberId, attrName => $A_LAST_NAME );
+		$users->{$login}->{$A_DISPLAY_NAME} = $data->getUserAttributeValue( member => $memberId, attrName => $A_DISPLAY_NAME );
+		$users->{$login}->{$A_USER_PREFERRED_MAIL} = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_PREFERRED_MAIL );
+
+		my $status = $data->getMemberAttributeValue( member => $memberId, attrName => $A_MEMBER_STATUS );
+		if ($status eq $STATUS_VALID) {
+			$status = 'ACTIVE';
+		} elsif ($status eq $STATUS_INVALID) {
+			$status = 'INCOMPLETE';
+		} elsif ($status eq $STATUS_EXPIRED) {
+			$status = 'INACTIVE';
+		} elsif ($status eq $STATUS_DISABLED) {
+			$status = 'ARCHIVED';
+		}
+		$users->{$login}->{$A_MEMBER_STATUS} = $status;
+
+		my $sshKey = $data->getUserAttributeValue( member => $memberId, attrName => $A_SSHKEY );
+		if (defined $sshKey and length $sshKey) {
+			$users->{$login}->{$A_SSHKEY} = $sshKey
+		}
+	}
+}
+
+# print BASE_DN file
+open FILE,">:encoding(UTF-8)","$baseDnFileName" or die "Cannot open $baseDnFileName: $! \n";
+print FILE $ldapBaseDN;
+close(FILE);
+
+# print user data LDIF
+open FILE,">:encoding(UTF-8)","$fileName" or die "Cannot open $fileName: $! \n";
+
+for my $login (sort keys %$users) {
+
+	print FILE "dn: uid=" . $login . "," . $ldapBaseDN . "\n";
+	print FILE "uid: " . $login . "\n";
+	print FILE "givenName: " . $users->{$login}->{$A_FIRST_NAME} . "\n";
+	print FILE "sn: " . $users->{$login}->{$A_LAST_NAME} . "\n";
+	print FILE "fullName: " . $users->{$login}->{$A_FIRST_NAME} . " " . $users->{$login}->{$A_LAST_NAME} . "\n";
+	print FILE "displayName: " . $users->{$login}->{$A_DISPLAY_NAME} . "\n";
+	print FILE "mail: " . $users->{$login}->{$A_USER_PREFERRED_MAIL} . "\n";
+	print FILE "userPassword: {SASL}" . $login . '@EINFRA' . "\n";
+	print FILE "state: " . $users->{$login}->{$A_MEMBER_STATUS} . "\n";
+
+	my $sshKey = $users->{$login}->{$A_SSHKEY};
+	if (defined $sshKey and length $sshKey) {
+		print FILE "sshPublicKey: " . $sshKey . "\n";
```

Comment:
SSH key attribute contains array. You must print it in a cycle on multiple lines. See other LDAP/AD services, where multi-value attributes are printed.
Also I think you can simplify original variable assignment when constructing `$users` structure. There is no necessity to check if its defined or non-empty, you can do it when printing the value here.

Reply:
Ok, thanks, I've also added sorting to compare in send script...